python/equistore-operations/equistore/operations/join.py

```diff
@@ -126,7 +160,22 @@ def join(tensors: List[TensorMap], axis: str):
     else:
         tensor_joined = tensor.keys_to_properties("tensor")
 
-    # TODO: once we have functions to manipulate meta data we can try to
-    # remove the `tensor` label entry after joining.
-
-    return tensor_joined
+    # Remove "tensor" Labels dimension if a new Labels of all blocks would be unique
+    # Due to toch script we can use a try-except block but have to check in advance if
+    # the `remove_dimension` would be succesful, i.e the new Labels instance is unique.
+    for block in tensor_joined:
+        if axis == "samples":
+            n_values = len(block.samples)
+            # "tensor" names is inserted at the end of the Labels dimensions
+            new_names = np.delete(block.samples, -1, axis=1)
+        elif axis == "properties":
+            n_values = len(block.properties)
+            # "tensor" name is inserted at the begininng of the Labels dimensions
+            new_names = np.delete(block.properties, 0, axis=1)
+
+        n_values_unique = len(np.unique(new_names, axis=0))
+
+        if n_values != n_values_unique:
+            return tensor_joined
```

Comment:
Perhaps `unique_metadata` can be used here?

```suggestion
    if len(equistore.unique_metadata(tensor_joined, axis=axis, name="tensor")) > 1:
        return tensor_joined
```

Reply:
Hmm apologies maybe not: see https://github.com/lab-cosmo/equistore/pull/314#discussion_r1273274348

Reply:
Yes as we talked yesterday I will try to use Labels::intersection.